pkg/types/nutanix/validation/platform.go

```diff
@@ -122,6 +124,22 @@ func ValidatePlatform(p *nutanix.Platform, fldPath *field.Path, c *types.Install
 		}
 	}
 
+	// validate the PreloadedOSImageName if configured
+	if p.PreloadedOSImageName != "" {
+		ctx, cancel := context.WithTimeout(context.TODO(), 60*time.Second)
+		defer cancel()
+
+		nc, err := nutanix.CreateNutanixClientFromPlatform(p)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(fldPath.Child("preloadedOSImageName"), fmt.Errorf("fail to create nutanix client. %w", err)))
```

Comment:
You should return here, since in case of errors `nc` will be `nil` and the next call will try to use the client.
```suggestion
			return append(allErrs, field.InternalError(fldPath.Child("preloadedOSImageName"), fmt.Errorf("fail to create nutanix client. %w", err)))
```

Reply:
fixed.